com.unity.probuilder/Editor/EditorCore/EditorSceneViewPicker.cs

```diff
@@ -106,7 +107,21 @@ public static ProBuilderMesh DoMouseClick(Event evt, SelectMode selectionMode, S
                     UndoUtility.RecordSelection(mesh, "Select Vertex");
 
                     if (ind > -1)
-                        mesh.SetSelectedVertices(mesh.selectedIndexesInternal.RemoveAt(ind));
+                    {
+                        var sharedIndex = mesh.sharedVertexLookup[s_Selection.vertex];
+                        var sharedVertex = mesh.sharedVerticesInternal[sharedIndex];
+                        s_IndexBuffer.Clear();
+                        foreach (var vertex in sharedVertex)
+                        {
+                            var index = Array.IndexOf(mesh.selectedIndexesInternal, vertex);
+                            if (index < 0)
+                                continue;
+
+                            s_IndexBuffer.Add(index);
+                        }
+                        
+                        mesh.SetSelectedVertices(mesh.selectedIndexesInternal.SortedRemoveAt(s_IndexBuffer));
```

Comment:
I don't think `selectedIndexesInternal` or `s_IndexBuffer` will be guaranteed to be ordered correctly.

Reply:
Now sorting indexes before passing them on to sorted remove at.